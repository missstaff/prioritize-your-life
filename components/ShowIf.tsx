const ShowIf = ({
  condition,
  render,
  renderElse = null,
}: {
  condition: boolean | undefined;
  render: React.ReactNode;
  renderElse?: React.ReactNode | null;
}) => {
  if (condition) {
    return <>{render}</>;
  }
  if (renderElse) {
    return <>{renderElse}</>;
  }
  return null;
};

export default ShowIf;
