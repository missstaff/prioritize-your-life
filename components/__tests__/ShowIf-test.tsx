import React from "react";
import { Text } from "react-native";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import ShowIf from "@/components/ShowIf";

describe("ShowIf Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ShowIf condition={true} render={<Text>Condition is true</Text>} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders the content when the condition is true", () => {
    const { getByText } = render(
      <ShowIf condition={true} render={<Text>Condition is true</Text>} />
    );

    expect(getByText("Condition is true")).toBeTruthy();
  });

  it("renders the alternative content when the condition is false and renderElse is provided", () => {
    const { getByText } = render(
      <ShowIf
        condition={false}
        render={<Text>Condition is true</Text>}
        renderElse={<Text>Condition is false</Text>}
      />
    );

    expect(getByText("Condition is false")).toBeTruthy();
  });

  it("does not render anything when the condition is false and renderElse is not provided", () => {
    const { queryByText } = render(
      <ShowIf condition={false} render={<Text>Condition is true</Text>} />
    );

    expect(queryByText("Condition is true")).toBeNull();
    expect(queryByText("Condition is false")).toBeNull();
  });

  it("renders nothing when condition is true but render is null", () => {
    const { queryByText } = render(
      <ShowIf
        condition={true}
        render={null}
        renderElse={<Text>Condition is false</Text>}
      />
    );

    expect(queryByText("Condition is false")).toBeNull();
  });
});
