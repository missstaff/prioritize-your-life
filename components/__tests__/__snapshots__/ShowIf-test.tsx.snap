// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowIf Component renders correctly 1`] = `
<Text>
  Condition is true
</Text>
`;
